### ISSUE TITLE
Fix ViewportTexture's "has_alpha()" returning false when the SubViewport it's from, has "transparent_bg" turned on.

### ISSUE THREAD

**Issue by KccbzZ** [godotengine/godot#94332]
### Tested versions

v4.2.2.stable.official [15073afe3]

### System information

Godot v4.2.2.stable - Windows 10.0.22631 - Vulkan (Forward+)

### Issue description

The **ViewportTexture** from a **SubViewport** with `transparent_bg` on, when calling `has_alpha()` on it, will always return _false_, despite `transparent_bg` being turned on.

Using `ViewportTexture.get_image()` is too resource-intensive for some use cases like mine, where I just have to differentiate pixels from one another in any way, from a texture created by a shader (To let the GPU handle the lag-inducing math that would normally occur on the CPU).

**Texture2D**'s `_is_pixel_opaque()` function is virtual, hence inaccessible.

I expect a **ViewportTexture**, when `transparent_bg` is on, to be able to have an alpha channel. This actually seems to be the case, as when calling `ViewportTexture.get_image().get_pixel(x: int, y: int).a` on a transparent pixel, it returns a value of 0 for the alpha, and for an opaque pixel, returns 1 for  the alpha. As a contradiction, `ViewportTexture.has_alpha()` returns _false_.

Using a **Sprite2D**'s `is_pixel_opaque(pos: Vector2)`, likewise, works if has_alpha() is _true_. And this is not the case for the **ViewportTexture** that gets returned.

Anything fast that can be called hundreds of times in GDScript that would allow fetching a pixel's data in a given **Texture2D**, in order to differentiate pixels from one another, would do for my use case.

### Steps to reproduce

1. Add a **SubViewport**
2. Add a **TextureRect** / **Sprite2D** as a child of it
3. Make the texture have transparent bits (I used a shader to do this in the example project, but same thing occurs with a png)
4. Use `get_texture()` on the **SubViewport**
5. On this **ViewportTexture**, call `has_alpha()`, which will be _false_
6. On this **ViewportTexture**, call `get_image().get_pixel(x:int, y:int)` on a pixel that is transparent. It will have an alpha value of 0, and the opaque bits will have an alpha value of 1, therefore contradicting step 5.

This, if the **ViewportTexture** is assigned to a **Sprite2D**'s texture, makes the **Sprite2D**'s `is_pixel_opaque(pos: Vector2)` unusable, as it will only return _true_.

### Minimal reproduction project (MRP)

[TextureViewport Alpha Issue.zip](https://github.com/user-attachments/files/16203699/TextureViewport.Alpha.Issue.zip)

**Comment by AThousandShips** [comment]
This is hard coded at the moment, so unsure how to treat that
https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/scene/main/viewport.cpp#L160-L162

**Comment by KccbzZ** [comment]
I'd love to help but unfortunately I haven't learned C++ yet.

Although, I'm thinking perhaps **Sprite2D**'s `is_pixel_opaque()` function could give some pointers, as once assigned a **ViewportTexture**, it'll return _true_ for every pixel it occupies.

https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/scene/2d/sprite_2d.cpp#L331-L380

**Comment by AThousandShips** [comment]
The viewport texture returns `false` for `is_pixel_opaque` by default for the same reason, this seems to be by design, which is why it should be discussed how to approach it and what is needed or desired here

**Comment by KccbzZ** [comment]
Well in terms of the desire, I initially used `get_distance_squared()` between 500 units simultaneously to allow them to push one another in a RTS prototype I'm developing, which caused FPS to drop to around 5, so the idea was to let the GPU handle this math by giving it an array of all units and the count of units.

This would be returned in texture format, but the only way to get that texture from the shader is using a **SubViewport**, to my knowledge at least, and what I've seen online.

So I found `is_pixel_opaque()` to be one of the few differentiators on a **Sprite2D**. In a test scene, I called `print(sprite.is_pixel_opaque()` in the `_process` function at probably more than 500 times on a texture that changes from `texture_1` to `texture_2` on a **Timer**, and it performed at 60 FPS.

Perhaps is there a way to execute math on the GPU in another way?

**Comment by KccbzZ** [comment]
> Actually fetching the pixel value from the texture is likely slow and expensive, so don't know if it is reasonable to implement it, instead I think this should be documented, and you should instead extract the image and get the data from there instead

This being expensive was correct, I just realized that in the test scene I described above I should've used the following to simulate my situation:

```gdscript
func _process(delta: float) -> void:
	for x in range(500):
		for y in range(500):
			print("\nOPAQUE: %s\nFPS: %s" % [is_pixel_opaque(Vector2(x, y)), Engine.get_frames_per_second()])
```

Using this, FPS is 1. But then given that this isn't a viable solution for such use cases, how else could this math be done quickly? I tried using Rect2's yesterday too.

**Comment by AThousandShips** [comment]
I don't think it can be done fast because it involves transferring data from the GPU, you could do stuff with a shader, see [here](https://docs.godotengine.org/en/latest/tutorials/shaders/screen-reading_shaders.html) and associated shader tutorials

**Comment by KccbzZ** [comment]
Understood. Well, if what I stated in the title is intended then I'm fine with having this issue closed, although I don't personally understand why `has_alpha()` would always be false for a **ViewportTexture**, given the existence of the `transparent_bg` property, but I'm sure that some additional documentation would be of help.

Thank you for the clarification!

**Comment by AThousandShips** [comment]
We should probably look at different options, including documenting this more clearly, so let's keep this open and see what's decided

**Comment by KccbzZ** [comment]
Sounds good. It would be wonderful to be able to do this math efficiently haha.
